### PR TITLE
feat(skore): Add a method to check performance consistency across cv splits

### LINKED ISSUE #1989: feat(skore): Detect high variance of metrics in cross valdiations
### Is your feature request related to a problem? Please describe.

It could be nice to warn users that a metric has a high variance in specific split of a cross validation.

### Describe the solution you'd like

Sample detection code by @glemaitre. 
```python
def detect_outliers_mad(scores, threshold=3.5):
    median = np.median(scores)
    mad = np.median(np.abs(scores - median))
    modified_z_scores = 0.6745 * (scores - median) / mad

    outliers = np.where(np.abs(modified_z_scores) > threshold)[0]
    return outliers, modified_z_scores
```

This new metric should also be serialized by `CrossValidationReportPayload`.

### Describe alternatives you've considered, if relevant

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by GaetandeCast**
Closes #1989.

Investigating how to detect cross-validation splits with unexpectedly good or bad performance using the Modified Z-score with Median Absolute Deviation (MAD).

Example usage:
```python
from sklearn.datasets import load_breast_cancer
from sklearn.linear_model import LogisticRegression
from skore import CrossValidationReport
X, y = load_breast_cancer(return_X_y=True)
classifier = LogisticRegression(max_iter=10_000)
report = CrossValidationReport(classifier, X=X, y=y, splitter=5)
report.check_split_consistency()
```

### CODE REVIEW

**Comment by GaetandeCast** [comment]
Closing this as we will probably revisit the feature within the auto diagnosis framework.
